keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceCheckHealthIndicator.kt

```diff
@@ -0,0 +1,36 @@
+package com.netflix.spinnaker.keel.actuation
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.actuate.health.AbstractHealthIndicator
+import org.springframework.boot.actuate.health.Health
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+import java.time.Clock
+import java.time.Duration
+import java.time.Instant
+import java.time.format.DateTimeFormatter.ISO_LOCAL_DATE_TIME
+
+@Component
+class ResourceCheckHealthIndicator(
+  private val clock: Clock,
+  @Value("\${keel.health.resource-check.threshold:PT1M}") private val healthyThreshold: Duration
+) : AbstractHealthIndicator() {
+
+  private var lastCheck: Instant = clock.instant()
+
+  override fun doHealthCheck(builder: Health.Builder) {
+    with(builder) {
+      if (lastCheck >= clock.instant() - healthyThreshold) {
+        up()
+      } else {
+        down()
```

Comment:
Won't this result in an `ApplicationDown` event that in-turn disables the `checkResources` actuation loop, which would in turn prevent this health check from ever recovering? If actuation cycles are missed for a minute due to a transient db or network issue, we'd want keel to recover on its own once that issue clears.